### PR TITLE
Replace npm with yarn

### PR DESCRIPTION
**Pull request by rarkins**
This PR _does not_ add `yarn` support as per #66. It's merely deploying `yarn` for this repository as a starting point.

### CODE REVIEW

**Comment by rarkins** [comment]
I'm holding off merging this until I decide on the best approach for pinning versus ranges in `package.json`, being discussed in #66.

**Comment by rarkins** [comment]
@jonbretman @hbetts any concerns on this change before merging? I thought it's good to dogfood the new `yarn` support from `renovate`.

**Review by jonbretman — APPROVED** [review]
Looks good to me.

File: circle.yml
Comment (hbetts) [inline circle.yml]: You may want to do `yarn install --frozen-lockfile`.

The `--frozen-lockfile` option forces `yarn` to ensure the `yarn.lock` file is valid given the project's `package.json` file.

The scenario where that flag is useful is if you, or someone else, where to `npm install` a new dependency, not using `yarn`, and then push the change to the repository. `--frozen-lockfile`, will fail the build early indicating the discrepancy between your `package.json` file, and your `yarn.lock` file.

Sadly, no documentation exists yet: yarnpkg/yarn#2478

Comment (rarkins) [inline circle.yml]: @hbetts Good idea

CircleCI runs v0.18.1 by default, so needed to manually upgrade it.

But then I noticed this problem, only fixed in `yarn` a day ago: https://github.com/yarnpkg/yarn/pull/2663

I'm considering whether to merge this anyway, and hope it gets included into v0.20.0 once it's released.

**Pull request merged**
